### PR TITLE
Story/qstr 99

### PR DESCRIPTION
**Pull request by AourpallyNikhil**
Merged develop into QSTR-99 and resolved conflicts.

### CODE REVIEW

File: QStore4S/pom.xml [hunk 1]
```diff
@@ -220,8 +220,8 @@
 			<!-- Dev Env. Properties -->
 			<properties>
 				<!-- Neo4j Properties -->
-				<neo4j.user>${neo4j.username}</neo4j.user>
-				<neo4j.pw>${neo4j.password}</neo4j.pw>
+				<neo4j.user>neo4j</neo4j.user>
+				<neo4j.pw>myneo4j</neo4j.pw>
```
Comment (jdamerow) [inline QStore4S/pom.xml:224]: Please put the variables back in here, or you will break the build for everybody else.

File: QStore4S/src/main/java/edu/asu/qstore4s/controller/QStore.java [hunk 2]
```diff
@@ -11,6 +12,7 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+//import org.springframework.security.core.Authentication;
```
Comment (jdamerow) [inline QStore4S/src/main/java/edu/asu/qstore4s/controller/QStore.java:15]: delete this line

File: QStore4S/src/main/webapp/WEB-INF/locale/messages_en_US.properties [hunk 3]
```diff
@@ -0,0 +1,2 @@
+label.user = User
+label.password = password
```
Comment (jdamerow) [inline QStore4S/src/main/webapp/WEB-INF/locale/messages_en_US.properties:2]: Probably capitalize "password". I assume it's the label for the password field?

File: QStore4S/src/main/webapp/WEB-INF/spring/spring-security.xml [hunk 4]
```diff
@@ -22,9 +22,7 @@
 	<headers />
 		<intercept-url pattern="/login*" access="permitAll" />
 		<intercept-url pattern="/logout*" access="permitAll" />
-		
-		<intercept-url pattern="/auth/welcome" access="hasAnyRole('ROLE_ADMIN')"/>
-
+		<intercept-url pattern="/auth/welcome" access="hasAnyRole('ROLE_ADMIN', 'ROLE_USER')"/>
```
Comment (jdamerow) [inline QStore4S/src/main/webapp/WEB-INF/spring/spring-security.xml:25]: just make this isAuthenticated as in line 26.

File: QStore4S/src/main/webapp/WEB-INF/views/login.jsp [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+			<tr>
+				<td colspan='2'><input name="submit" type="submit" value="Login" />
+				</td>
+			</tr>
+		</table>
+	</form>
+    </div> <!-- /container -->
+</span>
+</body>
+</html>
```
Comment (jdamerow) [inline QStore4S/src/main/webapp/WEB-INF/views/login.jsp:61]: The user should still be able to see the "up and running" page even if they are not logged in yet. Please add everything that was on the test page before to this page.

**Comment by jdamerow** [comment]
see comments

File: QStore4S/pom.xml (hunk 1 above)
Comment (AourpallyNikhil) [inline QStore4S/pom.xml:224]: Sure

**Comment by AourpallyNikhil** [comment]
All suggested changes made

**Pull request closed without merging**